twoliter/src/cmd/build.rs

```diff
@@ -23,12 +37,102 @@ pub(crate) struct BuildVariant {
+        let tempdir = tools_tempdir()?;
+        install_tools(&tempdir).await?;
+        let makefile_path = tempdir.path().join("Makefile.toml");
+        let packages_dir =
+            TempDir::new().context("Unable to create a tempdir for Twoliter's packages")?;
+
+        let sdk_container = DockerContainer::new(
+            "sdk",
+            project
+                .sdk(&self.arch)
+                .context("The project was missing an sdk")?
+                .uri(),
+        )
+        .await?;
+        sdk_container
+            .cp(Path::new("twoliter/alpha/build/rpms"), packages_dir.path())
+            .await?;
+
+        let rpms_dir = project.project_dir().join("build").join("rpms");
+        fs::create_dir_all(&rpms_dir)?;
+        debug!("Copying rpms to build dir");
+        for maybe_file in fs::read_dir(packages_dir.path().join("rpms"))? {
+            let file = maybe_file?;
+            if !file.file_type()?.is_file() {
+                debug!("Skipping '{}'", file.path().display());
+            }
+            debug!("Copying '{}'", file.path().display());
+            fs::copy(file.path(), rpms_dir.join(file.file_name()))?;
+        }
+
+        // Create the sbkeys directory
+        if !Path::new("sbkeys").is_dir() {
+            create_dir("sbkeys").await?;
+        }
+
+        let created_files = cp_alpha_files(&sdk_container).await?;
+
+        // Hold the result of the cargo make call so we can clean up the project directory first.
+        let res = CargoMake::new(&project, &self.arch)?
+            .env("TWOLITER_TOOLS_DIR", tempdir.path().display().to_string())
+            .env("GO_MODULES", &self.go_modules)
+            .env("BUILDSYS_ARCH", &self.arch)
+            .env("BUILDSYS_VARIANT", &self.variant)
+            .makefile(makefile_path)
+            .project_dir(project.project_dir())
+            .exec("build")
+            .await;
+
+        // Clean up all of the files we created
+        for file_name in created_files {
+            let added = Path::new(&file_name);
+            if added.is_file() {
+                remove_file(added).await?;
+            } else if added.is_dir() {
+                remove_dir_all(added).await?;
+            }
+        }
+
+        res
+    }
+}
+
+// Make sure the alpha build system has all required files
+async fn cp_alpha_files(sdk_container: &DockerContainer) -> Result<Vec<String>> {
+    let mut created_files = Vec::new();
+    for file_name in ALPHA_INFRA {
+        if Path::new(file_name).is_file() {
+            trace!("Alpha file '{file_name}' already exists. Skipping");
+            continue;
+        }
+        if Path::new(file_name).is_dir() {
+            trace!("Alpha directory '{file_name}' already exists. Skipping");
+            continue;
+        }
+        sdk_container
+            .cp(
+                &Path::new("twoliter/alpha/").join(file_name),
+                Path::new(file_name),
```

Comment:
Are we putting these into the user's project directory? If so that's wonky. Can we place them in a tempdir each time we are invoked?

Reply:
We are only adding a models directory to the users tree, and it is automatically cleaned up.